### ISSUE TITLE
Exception while trying to set key vault policy

### ISSUE THREAD

**Issue by espensb** [Azure/azure-cli#10512]
## Describe the bug

**Command Name**
`az keyvault set-policy`

**Errors:**
```
'NoneType' object is not subscriptable
Traceback (most recent call last):
python3.7/site-packages/knack/cli.py, ln 206, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 603, in execute
    raise ex
cli/core/commands/__init__.py, ln 661, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 654, in _run_job
    six.reraise(*sys.exc_info())
...
cli/command_modules/keyvault/custom.py, ln 214, in _get_object_id
    return _get_object_id_from_subscription(graph_client, subscription)
cli/command_modules/keyvault/custom.py, ln 197, in _get_object_id_from_subscription
    if subscription['user']:
TypeError: 'NoneType' object is not subscriptable
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- `az keyvault set-policy --secret-permissions get -n <vault-name>`

## Expected Behavior
Expected to be able to set policy

## Environment Summary
```
Linux-4.19.61-1-lts-x86_64-with-arch
Python 3.7.4
Shell: bash

azure-cli 2.0.73

```

**Comment by espensb** [comment]
I was missing the `--upn` argument here. It would still be nice with a proper error message instead of a crash here, though.

**Comment by yungezz** [comment]
Fixing of the issue will be released in next release 2.0.29, which will be released around Jan 10th.
